### DISCUSSION TITLE
werf cleanup with aws sso

### DISCUSSION THREAD

**Discussion by Bernix01** [werf/werf#6357] [General]
hello everyone, I am starting a discussion since it may not be a bug.
```
NoCredentialProviders:    ↵
│ │ │ │ no valid providers in chain. Deprecated.
│ │ │ │ 	For verbose messaging see aws.Config.CredentialsChainVerboseErrors
```
I can confirm aws access is ok with AWS CLI, I double checked cloud trail for any access error (none reported).

So I am guessing this is an error with aws-sdk-go. As title suggests, the credentials are set with SSO using sessions. I can also confirm that without sessions the error is the same.

I have also checked with debug and verbose enabled but no useful message is shown.
I am using the latest werf version.

For more context, `~/.aws/config` looks like this

```
[default]
sso_start_url = <url>
sso_region = <region>
sso_account_id = <account_id>
sso_role_name = <role_name>
region = <region>
output = json
```

**Comment by alexey-igrychev** [comment]
Thank you for the feedback. This is indeed our bug and limited support for AWS. You can read about the issue here and try the configuration version for SDK v1 ([link](https://one2n.io/blog/how-to-make-sso-auth-work-for-golang-aws-sdk)).

In [this PR](https://github.com/werf/werf/pull/6365), we transitioned to aws sdk v2, and everything should work as expected.

Let me know if you need any adjustments or further assistance!
